capirca/lib/arista_tp.py

```diff
@@ -0,0 +1,1036 @@
+                    src_str += " field-set src-%s" % self.term.name
+                else:
+                    for addr in src_addr:
+                        src_str += " %s" % addr
+
+                term_block.append([MATCH_INDENT, src_str, False])
+            elif self.term.source_address:
+                logging.debug(
+                    self.NO_AF_LOG_ADDR.substitute(
+                        term=self.term.name, direction="source",
+                        af=self.term_type
+                    )
+                )
+                return ""
+
+            # destination address
+            dst_addr = self.term.GetAddressOfVersion("destination_address",
+                                                     term_af)
+            dst_addr_ex = self.term.GetAddressOfVersion(
+                "destination_address_exclude", term_af
+            )
+
+            if dst_addr:
+                dst_str = "destination prefix"
+                if dst_addr_ex:
+                    # this should correspond to the generated field set
+                    dst_str += " field-set dst-%s" % self.term.name
+                else:
+                    for addr in dst_addr:
+                        dst_str += " %s" % addr
+
+                term_block.append([MATCH_INDENT, dst_str, False])
+
+            elif self.term.destination_address:
+                logging.debug(
+                    self.NO_AF_LOG_ADDR.substitute(
+                        term=self.term.name, direction="destination",
+                        af=self.term_type
+                    )
+                )
+                return ""
+
+            if self.term.source_prefix:
+                src_pfx_str = "source prefix field-set"
+                for pfx in self.term.source_prefix:
+                    src_pfx_str += " %s" % pfx
+
+                term_block.append([MATCH_INDENT,
+                                   " %s" % src_pfx_str, False])
+
+            if self.term.destination_prefix:
+                dst_pfx_str = "destination prefix field-set"
+                for pfx in self.term.destination_prefix:
+                    dst_pfx_str += " %s" % pfx
+
+                term_block.append([MATCH_INDENT,
+                                   " %s" % dst_pfx_str, False])
+
+            # PROTOCOL MATCHES
+            protocol_str = ""
+            if self.term.protocol:
+                protocol_str = self._processProtocol(self.term_type,
+                                                     self.term, flags)
+
+            # protocol-except handling
+            if self.term.protocol_except:
+                protocol_str = self._processProtocolExcept(self.term_type,
+                                                           self.term, flags)
+
+            # tcp/udp port generation
+            port_str = self._processPorts(self.term)
+            if port_str != "":
+                protocol_str += port_str
+
+            # icmp[v6] handling
```

Comment:
You'll want to add a check here to ensure that icmp is only rendered when the term contains actual IPv4 addresses only, and icmpv6 is rendered when there's only IPv6 addresses.